src/modules/notification/infrastructure/store/types/index.ts

```diff
@@ -0,0 +1,62 @@
+import EventNotification from '@/modules/notification/domain/entity/EventNotification';
+import {
+  ActionContext,
+  Store as VuexStore,
+  CommitOptions,
+  DispatchOptions
+} from 'vuex';
+
+export enum MutationTypes {
+  SET_NOTIFICATIONS = 'SET_NOTIFICATIONS'
+}
+
+export enum ActionTypes {
+  SET_NOTIFICATIONS = 'SET_NOTIFICATIONS'
+}
+
+export type State = {
+  all: EventNotification[]
+};
+
+export type Mutations<S = State> = {
+  [MutationTypes.SET_NOTIFICATIONS](state: S, payload: EventNotification[]): void;
+};
+
+export type Getters = {
+  notificationList(satate: State): EventNotification[]
+};
+
+type AugmentedActionContext = {
+  commit<K extends keyof Mutations>(
+    key: K,
+    payload: Parameters<Mutations[K]>[1]
+  ): ReturnType<Mutations[K]>
+} & Omit<ActionContext<State, State>, 'commit'>;
+
+export interface Actions {
+  [ActionTypes.SET_NOTIFICATIONS](
+    { commit }: AugmentedActionContext,
+    payload: EventNotification[]
+  ): void;
+}
+
+export type Store = Omit<
+  VuexStore<State>,
+  'getters' | 'commit' | 'dispatch'
+> & {
+  commit<K extends keyof Mutations, P extends Parameters<Mutations[K]>[1]>(
+    key: K,
+    payload: P,
+    options?: CommitOptions
+  ): ReturnType<Mutations[K]>
+} & {
+  dispatch<K extends keyof Actions>(
+    key: K,
+    payload: Parameters<Actions[K]>[1],
+    options?: DispatchOptions
+  ): ReturnType<Actions[K]>
+} & {
+  getters: {
+    [K in keyof Getters]: ReturnType<Getters[K]>
+  }
+};
```

Comment:
I don't remember having to define these types by ourselves when working in Twake Console. Aren't they supposed to be inferred from Vuex's provided types?

Reply:
Vuex 4 types are not specific enough, so we had to create a fully statically typed store with some intersection with Vuex types.

more details: https://dev.to/3vilarthas/vuex-typescript-m4j